src/Tape.cpp

```diff
@@ -712,13 +711,13 @@ void Tape<T>::clearDerivativesAfter(position_type pos)
 }
 
 template <class T>
-T Tape<T>::getAndResetOutputAdjoint(slot_type slot)
-{
-    if (slot >= slot_type(derivatives_.size()))
+T Tape<T>::getAndResetOutputAdjoint(slot_type slot) {
+    const auto derivatives_size = derivatives_.size();
+    if (slot >= slot_type(derivatives_size))
         throw OutOfRange("Requested output slot does not exist");
 
-    T ret = derivatives_[slot];
-    derivatives_[slot] = 0.0;
+    T ret = std::move(derivatives_[slot]);  // use move semantics
+    derivatives_[slot] = T();  // more efficient
```

Comment:
Good idea here - it makes a difference for higher order derivatives (if T is not just double). Note that for doubles, move has no advantage over copies, it's the same.

Could you remove that comment in the assignment ("more efficient")? Because that isn't actually more efficient, it's more generic.

Reply:
yeah true, i just updated